File: Android_App/app/src/main/java/uk/ac/aber/dcs/cascade/ui/database/enitity/Module.kt
package uk.ac.aber.dcs.cascade.ui.database.enitity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.jetbrains.annotations.NotNull

/**
 * @author Philip
 */
@Entity
data class Module(
    @PrimaryKey @ColumnInfo(name = "module_id")val moduleId: String,
    val title: String,
    val coordinator: String,
    val semester: String,
    val year: String,
    val department: String,
    val credits: Int,
    val welsh: Boolean,
    val url: String
)

File: AberCascadeCalulator/Models/Module.cs
using System;
using System.Collections.Generic;

namespace AberCascadeCalulator.Models
{
    public class Module
    {
        public string ModuleId { get; set; }
        public string Title { get; set; }
        public string Coordinator { get; set; }
        public string Semester { get; set; }
        public string Year { get; set; }
        public string Department { get; set; }
        public long? Credits { get; set; }
        public bool Welsh { get; set; }
        public string Url { get; set; }
    }
}
